server/src/main/scala/org/apache/spark/sql/crossdata/session/HazelcastSessionProvider.scala

```diff
@@ -15,116 +15,87 @@
  */
 package org.apache.spark.sql.crossdata.session
 
-import java.util
-
-import com.hazelcast.config.{Config => HazelcastConfig}
+import com.hazelcast.config.{GroupConfig, XmlConfigBuilder, Config => HazelcastConfig}
 import com.hazelcast.core.Hazelcast
-import com.stratio.crossdata.server.config.ServerConfig
 import com.typesafe.config.Config
 import org.apache.spark.SparkContext
 import org.apache.spark.sql.SQLConf
 import org.apache.spark.sql.crossdata.XDSessionProvider.SessionID
-import org.apache.spark.sql.crossdata.catalog.inmemory.HashmapCatalog
 import org.apache.spark.sql.crossdata.catalog.interfaces.XDTemporaryCatalog
 import org.apache.spark.sql.crossdata.{XDSession, XDSessionProvider, XDSessionState, XDSharedState}
 
-import scala.collection.JavaConversions._
 import scala.util.{Failure, Success, Try}
 
 object HazelcastSessionProvider {
 
-  // val SessionListId = "sessions"
   val SqlConfMapId = "sqlconfmap"
+  val HazelcastCatalogMapId = "hazelcatalogmap"
+
+  // TODO this is not the right place
+  def checkNotNull[T]: T => Try[T] =
+    a => Option(a).map(Success(_)).getOrElse(Failure(new RuntimeException(s"Map not found")))
 
 }
 
-class HazelcastSessionProvider(
-                                @transient sc: SparkContext,
+class HazelcastSessionProvider( @transient sc: SparkContext,
                                 userConfig: Config
                                 ) extends XDSessionProvider(sc, Option(userConfig)) {
 
   import HazelcastSessionProvider._
-  import ServerConfig._
+  import XDSharedState._
 
-  private val sharedState = new XDSharedState(sc, Option(userConfig))
+  private val sharedState = new XDSharedState(sc, Option(userConfig))// TODO add sqlConf
 
   // TODO manage config
-  private val cfg = new HazelcastConfig
-  private val hInstance = Hazelcast.newHazelcastInstance(cfg)
+  private val hInstance = {
+    val xmlConfig = new XmlConfigBuilder().build()
+    xmlConfig.setGroupConfig(new GroupConfig(scala.util.Properties.scalaPropOrElse("version.number", "unknown")))
```

Comment:
It is required temporarily because of a CD issue and should be fixed before merging session branch. I'll make the TODO more explicit